panther_analysis_tool/lookup_table.py

```diff
@@ -0,0 +1,171 @@
+"""
+Panther Analysis Tool is a command line interface for writing,
+testing, and packaging policies/rules.
+Copyright (C) 2020 Panther Labs Inc
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+import json
+import logging
+
+import requests
+
+from panther_analysis_tool.util import get_client
+
+LOOKUP_LAMBDA_NAME = "panther-lookup-tables-api"
+
+class LookupTable():
+    """A Panther Lookup Table"""
+
+    def __init__(self, spec: dict, spec_dir: str, aws_profile: str):
+        """Instantiate a Lookup Table object
+
+        Args:
+            spec: a dict representing the configuration of a Lookup Table
+        """
+        self.name = spec["name"]
+        self.spec = spec
+        self.spec_dir = spec_dir
+        self.aws_profile = aws_profile
+        self.client = get_client(self.aws_profile, "lambda")
+        self.lookup_id = None
+        self.input_bucket = ""
+        logging.info("Creating/updating the lookup table %s", self.name)
+
+
+    def update(self) -> int:
+        """Create a new Lookup Table or update a pre-existing one in Panther
+
+        Returns 1 if the operation fails
+
+        Args: none
+
+        Returns:
+        An integer return code
+        """
+        logging.info("Creating or updating the lookup table specified in %s", self.spec["dataFile"])
+
+        # First upload the data file to S3
+        upload_url = self.upload_data()
+        if not upload_url:
+            return 1
+
+        request_payload = json.dumps(
```

Comment:
[optional] this would be easier to read if you move the `json.dumps()` into the invoke statement and made this a dictionary with a single field per line

Reply:
I did that so that I could add the logging.debug() line. Can remove if you don't think that helps. @lindsey-w

Reply:
@lindsey-w I considered adding more tests but as unit tests, we can't really test much with this code other than that it parses responses output correctly. Also, I have integration tests in the backend which cover the API functionality.

It would be nice to have a framework in place that ran PAT tests against an actual backend but it would take some effort to build that out.